lib/compress/zstd_double_fast.c

```diff
@@ -409,9 +409,9 @@ static size_t ZSTD_compressBlock_doubleFast_extDict_generic(
         hashSmall[hSmall] = hashLong[hLong] = curr;   /* update hash table */
 
         if ((((U32)((prefixStartIndex-1) - repIndex) >= 3) /* intentional underflow : ensure repIndex doesn't overlap dict + prefix */
-            & (repIndex > dictStartIndex))
+            & (offset_1 < curr+1 - dictStartIndex)) /* note: we are searching at curr+1 */
           && (MEM_read32(repMatch) == MEM_read32(ip+1)) ) {
-            const BYTE* repMatchEnd = repIndex < prefixStartIndex ? dictEnd : iend;
+            const BYTE* repMatchEnd = offset_1 > curr+1 - prefixStartIndex ? dictEnd : iend;
```

Comment:
This shouldn't be necessary to change, here and elsewhere. We should only need to change the comparison on like 412. If the comparison succeeds, then we know it won't underflow.

Also `repBase` and `repMatch` are calculated using `repIndex` so it is a bit confusing to see `repMatchEnd` calculated differently.